src/scripts/generate-function-call.py

```diff
@@ -0,0 +1,134 @@
+        test_case_in_a_file = 70 # The number of test cases before new file begins
+        test_cases = ['args', 'call', 'apply', 'bind', 'recursive'] # The possible test cases
+        # The following lists are sublists of possible test cases
+        args_and_bind = ['args', 'bind']
+        call_and_apply = ['call', 'apply']
+
+    def concat_list_elements(self, list):
+        return ''.join(map(str, list))
+
+    def create_file_header(self): # Creates the header of the generated files
+        return self.concat_list_elements([validate.validate_numeric,
+                                         validate.test_function,
+                                         validate.test_function2,
+                                         '\nvar a = test_function;\n'])
+
+    def append_test_case(self): # Appends test_case to file output
+            self.file_output.append(self.test_case)
+            self.test_case = ''
+
+    def configure_settings(self, options): # Adjust file_write_settings based on options
+        return settings.file_write_settings(self._InnerValues.generated_filename,
+                                            options.output,
+                                            self.create_file_header(),
+                                            self._InnerValues.test_case_in_a_file)
+
+    def specify_test_and_expected_value(self, op): # Creates test's- and expected value's string
+        # specify the number of arguments here
+        if op not in self._InnerValues.args_and_bind:
+            number_of_args = random.randint(self._InnerValues.args_min, self._InnerValues.args_max - 1)
+        else:
+            number_of_args = random.randint(self._InnerValues.args_min, self._InnerValues.args_max)
+        if op != 'recursive':
+            self.expected_value = 'true'
+            # Writes 1 as argument for number_of_args times
+            self.test_value = ''
+            for j in range(number_of_args):
+                self.test_value += '1'
+                if j != number_of_args - 1:
+                    self.test_value += ', '
+            # Creates the string of the function call
+            if op in self._InnerValues.args_and_bind:
+                self.test_value = '(%s)' % (self.test_value)
+            elif op == 'apply':
+                self.test_value = '[%s]' % (self.test_value)
+            if op in self._InnerValues.call_and_apply + ['bind']:
+                if op in self._InnerValues.call_and_apply:
+                    self.test_value = 'this, %s' % (self.test_value)
+                    self.test_value = '(%s)' % (self.test_value)
+                else:
+                    self.test_value = '(this)%s' % (self.test_value)
+                self.test_value = '.%s%s' % (op, self.test_value)
+            if (random.randint(0, 1)):
+                self.test_value = 'test_function%s' % (self.test_value)
+            else:
+                self.test_value = 'a%s' % (self.test_value)
```

Comment:
What is that `a` prefix? Please clarify it.

Reply:
'a' is a reference to this declaration:
var a = test_function;